presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViews.java

```diff
@@ -295,6 +297,45 @@ public void testSelectFromHiveViewWithoutDefaultCatalogAndSchema()
                 .containsExactly(row(1L));
     }
 
+    @Test
+    public void testTimestampHiveView()
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS timestamp_hive_table");
+        onHive().executeQuery("CREATE TABLE timestamp_hive_table AS SELECT cast('1990-01-02 12:13:14.123456789' AS timestamp) ts");
+        onHive().executeQuery("DROP VIEW IF EXISTS timestamp_hive_view");
+        onHive().executeQuery("CREATE VIEW timestamp_hive_view AS SELECT * FROM timestamp_hive_table");
+
+        // timestamp_precision not set
+        unsetSessionProperty("hive.timestamp_precision");
+        unsetSessionProperty("hive_timestamp_nanos.timestamp_precision");
+
+        assertThat(query("SELECT CAST(ts AS varchar) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"));
+        assertThatThrownBy(
+                // TODO(https://github.com/prestosql/presto/issues/6295) it is not possible to query Hive view with timestamps if hive.timestamp-precision=NANOSECONDS
+                () -> assertThat(query("SELECT CAST(ts AS varchar) FROM hive_timestamp_nanos.default.timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123456789"))
+        ).hasMessageContaining("timestamp(9) projected from query view at position 0 cannot be coerced to column [ts] of type timestamp(3) stored in view definition");
+
+        setSessionProperty("hive.timestamp_precision", "'MILLISECONDS'");
+        setSessionProperty("hive_timestamp_nanos.timestamp_precision", "'MILLISECONDS'");
+
+        assertThat(query("SELECT CAST(ts AS varchar) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"));
+        assertThatThrownBy(
+                // TODO(https://github.com/prestosql/presto/issues/6295) it is not possible to query Hive view with timestamps if hive.timestamp-precision=NANOSECONDS
+                () -> assertThat(query("SELECT CAST(ts AS varchar) FROM hive_timestamp_nanos.default.timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"))
+        ).hasMessageContaining("timestamp(9) projected from query view at position 0 cannot be coerced to column [ts] of type timestamp(3) stored in view definition");
```

Comment:
i guess code style (& checkstyle) expects `)` to be on prev line

Reply:
Acually no (at least for checkstyle).

As for the codestyle:
```
        assertThatThrownBy(
                // TODO(https://github.com/prestosql/presto/issues/6295) it is not possible to query Hive view with timestamps if hive.timestamp-precision=NANOSECONDS
                () -> assertThat(query("SELECT CAST(ts AS varchar) FROM hive_timestamp_nanos.default.timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123456789")))
                .hasMessageContaining("timestamp(9) projected from query view at position 0 cannot be coerced to column [ts] of type timestamp(3) stored in view definition");
```
looks terrible IMO